flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -3,27 +3,257 @@ import Sentry
+                            event.sdk = ["integrations":integrations]
+                        }
+                    }
+                }
+                return event
+            }*/
+        }
+
+        result("iOS initNativeSdk" )
+    }
+
+    private func setEventOriginTag(event: Event){
+        guard let sdk = event.sdk else { return  }
+        if self.isValidSdk(sdk:sdk){
+
+            switch sdk["name"] as! String{
+            case "sentry.dart.flutter":
+                setEventEnvironmentTag(event:event,origin:"flutter",environment:"dart")
+            case "sentry.cocoa":
+                setEventEnvironmentTag(event:event,origin: "flutter",environment: "dart")
+            case "sentry.native":
+                setEventEnvironmentTag(event:event,origin: "flutter",environment: "dart")
+            default:
+                return
+            }
+        }
+    }
+
+    private func setEventEnvironmentTag(event: Event, origin: String = "ios", environment: String) {
+        event.tags?["event.origin"] = origin
+        event.tags?["event.environment"] = environment
+    }
+
+    private func isValidSdk( sdk: [String: Any]) -> Bool{
+        return (sdk["name"] != nil && !(sdk["name"] as! String).isEmpty)
+    }
+
+    func convertStringToDictionary(text: String) -> [String:AnyObject]? {
+        if let data = text.data(using: .utf8) {
+            do {
+                let json = try JSONSerialization.jsonObject(with: data, options: .mutableContainers) as? [String:AnyObject]
+                return json
+            } catch {
+                print("Something went wrong")
+            }
+        }
+        return nil
+    }
+
+    private func captureEnvelope(_ call: FlutterMethodCall, result: @escaping FlutterResult){
+        guard let arguments = call.arguments as? [Any],
+              !arguments.isEmpty,
+              let event = arguments.first as? String else {
+            result(FlutterError(code: "2", message: "Envelope is null or empty", details: nil) )
+            return
+        }
+
+        let parts = event.split(separator: "\n")
+        let envelopeParts: [[String: Any]] = parts.map ({ part in
+            convertStringToDictionary(text: "\(part)")!
+        })
+
+        let envelopeHeaderDict = envelopeParts[0]
+        let eventSdk = envelopeHeaderDict
+        let itemHeader = envelopeParts[1]
+
+        guard
+            let eventId = envelopeHeaderDict["event_id"] as? String,
+            let itemType = itemHeader["type"] as? String else {
+            result(FlutterError(code: "2", message: "Cannot serialize event", details: nil) )
+            return
+        }
+
+        let sdkInfo = SentrySdkInfo(dict: eventSdk)
+        let sentryId = SentryId(uuidString: eventId)
+        let envelopeHeader = SentryEnvelopeHeader.init(id: sentryId, andSdkInfo: sdkInfo)
+
+        let payload = envelopeParts[2]
+
+        let data = try! JSONSerialization.data(withJSONObject: payload, options: .init(rawValue: 0))
```

Comment:
if this fail, let's log it and return result(error), or what does it happen?

Reply:
✅